### PR TITLE
[SPARK-50389][SQL][TESTS] Remove`sql-tests/results/try_arithmetic.sql.out.java21`

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?

There are 8 Java 21-related golden files in Spark:

```
./sql/core/src/test/resources/sql-tests/results/try_arithmetic.sql.out.java21
./sql/core/src/test/resources/sql-tests/results/postgreSQL/int8.sql.out.java21
./sql/core/src/test/resources/sql-tests/results/postgreSQL/float4.sql.out.java21
./sql/core/src/test/resources/sql-tests/results/postgreSQL/numeric.sql.out.java21
./sql/core/src/test/resources/sql-tests/results/nonansi/try_arithmetic.sql.out.java21
./sql/core/src/test/resources/sql-tests/results/nonansi/try_aggregates.sql.out.java21
./sql/core/src/test/resources/sql-tests/results/try_aggregates.sql.out.java21
./sql/core/src/test/resources/sql-tests/results/datetime-formatting.sql.out.java21
```

After SPARK-50313 was merged, the contents of `./sql/core/src/test/resources/sql-tests/results/try_arithmetic.sql.out.java21` and `./sql/core/src/test/resources/sql-tests/results/try_arithmetic.sql.out` have become identical. We can verify by running `diff ./sql/core/src/test/resources/sql-tests/results/try_arithmetic.sql.out.java21 ./sql/core/src/test/resources/sql-tests/results/try_arithmetic.sql.out`. So this pr deletes this file.

Actually, after SPARK-50313 the specialization for Java 21 has become `./sql/core/src/test/resources/sql-tests/results/nonansi/try_arithmetic.sql.out.java21`.

### Why are the changes needed?
Delete an unnecessary golen file.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
- Pass Github Actions

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ on:
       java:
         required: false
         type: string
-        default: 17
+        default: 21
```
Comment (LuciferYang) [inline .github/workflows/build_and_test.yml:28]: will revert after test

**Comment by LuciferYang** [comment]
cc @dongjoon-hyun and @yaooqinn

**Review by dongjoon-hyun — APPROVED** [review]
Thank you for double-checking, @LuciferYang . Yes, +1 for removal.
(After recovering build_and_test.yml, feel free to merge).

**Comment by LuciferYang** [comment]
Thanks @dongjoon-hyun

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by LuciferYang** [comment]
Thanks @dongjoon-hyun

**Comment by dongjoon-hyun** [comment]
Merged to master~

**Comment by yaooqinn** [comment]
late lgtm

**Pull request closed without merging**